### DISCUSSION TITLE
linebreaks

### DISCUSSION THREAD

**Discussion by xiaoniuniu89** [markdoc/markdoc#338] [Q&A] [answered]
Hello, Is there a way to recognize single line breaks when using markdoc? Thank you.

**Comment by rpaul-stripe** [comment, marked as answer]
Yes, Markdoc automatically inserts a ["softbreak" node](https://github.com/markdoc/markdoc/blob/main/src/schema.ts#L230) in the document AST in places where there is a single line break. By default, a softbreak renders as a single space, but you can override that behavior by providing a custom node definition schema. You can also programmatically detect softbreaks by walking the AST and looking for nodes where the type is "softbreak". Here's an example:

```js
const example = `
# Example

This is a test
of a paragraph with a softbreak
`;

const config = {
  nodes: {
    softbreak: {
      transform() {
        // Custom behavior here
      }
    }
  }
};

const doc = Markdoc.parse(example);

for (const node of doc.walk())
  if (node.type === 'softbreak')
    console.log(`Found a softbreak in paragraph at line ${node.lines[0]}`)

const output = Markdoc.transform(doc, config);
```

**Reply by xiaoniuniu89** [reply]
perfect, thank you so much, that is very helpful!

**Reply by eerison** [reply]
Hey @rpaul-stripe , what should I put inside of `transform()` ? `return '</ br>';`? it isn't parsing html :'(

**Reply by rpaul-stripe** [reply]
@eerison the `transform` function typically returns a [`RenderableTree`](https://github.com/markdoc/markdoc/blob/main/src/types.ts#L96) which consists of [`Tag`](https://github.com/markdoc/markdoc/blob/main/src/tag.ts#L3) objects and plain text strings. To make it output a `br` HTML tag instead of a space, you would have the function return: `new Markdoc.Tag('br')`

**Reply by eerison** [reply]
Thank you 🥰
